src/lib/base.jl

```diff
@@ -138,3 +138,7 @@ end
     end
     return getfield(p, i), pair_getfield
 end
+
+@adjoint Base.nameof(x::UnionAll) = nameof(x), _ -> (nothing,)
+
+@nograd typeintersect
```

Comment:
Should these adjoints be moved to ChainRules?

Reply:
I don't see much precedence in ChainRules for these kinds of methods. Is that judgement correct @willtebbutt ?

Either way is fine, of course. We'll need them to emit `rrules` soon.

Reply:
We would very much like to have this in `ChainRules` -- it's actually precisely this kind of thing that we want to cover really thoroughly, there's just a lot of cover :)